### PR TITLE
Integration of hashicorp-vault-pipeline-plugin

### PR DESCRIPTION
**Pull request by jetersen**

### CODE REVIEW

**Comment by rcomanne** [comment]
Any update on this PR? What is required to merge this?

**Comment by casz** [comment]
I was exploring how to properly hide secrets 😓

**Comment by rcomanne** [comment]
> I was exploring how to properly hide secrets 😓

Ah okay, I did some quick testing with that as well, but it seemed like it requires a complete rewrite of the current step, as the others I have found, are all build wrappers so they keep control over the context, while our step just injects some values in the environment.
It is possible to combine this step with the withCredentials wrapper that will hide the secrets, so that is how we use the plugin atm.
I did not see a better quick way to implement this, so good luck and looking forward to how you'll solve this!

**Comment by rcomanne** [comment]
Hi @casz, any updates on this PR? Are we still waiting on masking passwords?
We can still use the functionality, users will just have to use the MaskPasswords plugin.

File: README.md [hunk 1]
```diff
@@ -125,6 +125,69 @@ node {
...
+        stage("read vault key") {
+            steps {
+                echo "${SECRET}"
+            }
+        }
+    }
+}
+```
+
+##### Masking secrets in console output
```
Comment (timja) [inline README.md]: In azure-keyvault plugin I used a block scoped step not the environment option and secrets were hidden fine:
https://github.com/jenkinsci/azure-keyvault-plugin#declarative

**Comment by vyas-n** [comment]
Any updates for these changes? The current https://github.com/jenkinsci/hashicorp-vault-pipeline-plugin is broken

**Comment by jetersen** [comment]
No longer relevant after https://github.com/jenkinsci/hashicorp-vault-plugin/pull/174

**Comment by vyas-n** [comment]
Thanks @jetersen, I didn't know #174 was an option!

If anyone finds this in the future, the PR above makes the `withVault` syntax mask credentials for pipelines.

Usage docs: https://www.jenkins.io/doc/pipeline/steps/hashicorp-vault-plugin
Example from README.md: https://github.com/jenkinsci/hashicorp-vault-plugin#usage-via-jenkinsfile

**Pull request closed without merging**